pkg/ui/markdown/styles.go

```diff
@@ -0,0 +1,272 @@
+	defaultBytes, err := getBuiltinDefaultStyle()
+	if err != nil {
+		return nil, err
+	}
+
+	if atmosConfig.Settings.Markdown.Document.Color == "" &&
+		atmosConfig.Settings.Markdown.Heading.Color == "" &&
+		atmosConfig.Settings.Markdown.H1.Color == "" &&
+		atmosConfig.Settings.Markdown.H2.Color == "" &&
+		atmosConfig.Settings.Markdown.H3.Color == "" {
+		return defaultBytes, nil
+	}
+
+	var style ansi.StyleConfig
+	if err := json.Unmarshal(defaultBytes, &style); err != nil {
+		return nil, err
+	}
+
+	// Apply custom styles on top of defaults
+	if atmosConfig.Settings.Markdown.Document.Color != "" {
+		style.Document.Color = &atmosConfig.Settings.Markdown.Document.Color
+	}
+
+	if atmosConfig.Settings.Markdown.Heading.Color != "" {
+		style.Heading.Color = &atmosConfig.Settings.Markdown.Heading.Color
+		style.Heading.Bold = &atmosConfig.Settings.Markdown.Heading.Bold
+	}
+
+	if atmosConfig.Settings.Markdown.H1.Color != "" {
+		style.H1.Color = &atmosConfig.Settings.Markdown.H1.Color
+		if atmosConfig.Settings.Markdown.H1.BackgroundColor != "" {
+			style.H1.BackgroundColor = &atmosConfig.Settings.Markdown.H1.BackgroundColor
+		}
+		style.H1.Bold = &atmosConfig.Settings.Markdown.H1.Bold
+		style.H1.Margin = uintPtr(uint(atmosConfig.Settings.Markdown.H1.Margin))
+	}
+
+	if atmosConfig.Settings.Markdown.H2.Color != "" {
+		style.H2.Color = &atmosConfig.Settings.Markdown.H2.Color
+		style.H2.Bold = &atmosConfig.Settings.Markdown.H2.Bold
+	}
+
+	if atmosConfig.Settings.Markdown.H3.Color != "" {
+		style.H3.Color = &atmosConfig.Settings.Markdown.H3.Color
+		style.H3.Bold = &atmosConfig.Settings.Markdown.H3.Bold
+	}
+
+	if atmosConfig.Settings.Markdown.CodeBlock.Color != "" {
+		if style.CodeBlock.StyleBlock.StylePrimitive.Color == nil {
+			style.CodeBlock.StyleBlock.StylePrimitive.Color = &atmosConfig.Settings.Markdown.CodeBlock.Color
+		} else {
+			*style.CodeBlock.StyleBlock.StylePrimitive.Color = atmosConfig.Settings.Markdown.CodeBlock.Color
+		}
+		style.CodeBlock.Margin = uintPtr(uint(atmosConfig.Settings.Markdown.CodeBlock.Margin))
+	}
+
+	if atmosConfig.Settings.Markdown.Link.Color != "" {
+		style.Link.Color = &atmosConfig.Settings.Markdown.Link.Color
+		style.Link.Underline = &atmosConfig.Settings.Markdown.Link.Underline
+	}
+
+	if atmosConfig.Settings.Markdown.Strong.Color != "" {
+		style.Strong.Color = &atmosConfig.Settings.Markdown.Strong.Color
+		style.Strong.Bold = &atmosConfig.Settings.Markdown.Strong.Bold
+	}
+
+	if atmosConfig.Settings.Markdown.Emph.Color != "" {
+		style.Emph.Color = &atmosConfig.Settings.Markdown.Emph.Color
+		style.Emph.Italic = &atmosConfig.Settings.Markdown.Emph.Italic
+	}
+
+	return json.Marshal(style)
+}
+
+// this only returns the built-in default style configuration
+func getBuiltinDefaultStyle() ([]byte, error) {
+	style := ansi.StyleConfig{
+		Document: ansi.StyleBlock{
+			StylePrimitive: ansi.StylePrimitive{
+				BlockPrefix: "",
+				BlockSuffix: "\n",
+				Color:       stringPtr("#FFFFFF"),
```

Comment:
Can we do some named colors like this or other conventional way? I don't care too much on how we do it, just prefer we don't embed hex codes everywhere. Hard to establish a color pallet. 

```
// colors/colors.go
package colors

import "github.com/charmbracelet/lipgloss"

// Define your custom named colors
var (
    White  = lipgloss.Color("#FFFFFF")
    Purple = lipgloss.Color("#9B51E0")
    Blue   = lipgloss.Color("#00A3E0")
)
```

and then do:

```
func getBuiltinDefaultStyle() ([]byte, error) {
    style := ansi.StyleConfig{
        Document: ansi.StyleBlock{
            StylePrimitive: ansi.StylePrimitive{
                BlockPrefix: "",
                BlockSuffix: "\n",
                Color:       stringPtr(colors.White.String()),
            },
            Margin: uintPtr(0),
        },

...
```